core/jvm/src/test/scala/scalaz/zio/RetrySpec.scala

```diff
@@ -119,29 +124,50 @@ class RetrySpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRun
   }
 
   def retryN = {
-    val retried  = unsafeRun(retryCollect(IO.fail("Error"), Schedule.recurs(5)))
-    val expected = (Left("Error"), List(1, 2, 3, 4, 5, 6).map((Duration.Zero, _)))
+    val retried = unsafeRun(retryCollect(IO.fail("Error"), Schedule.recurs(5)))
+    val expected =
+      (Left("Error"), List(1, 2, 3, 4, 5, 6).map((Delay.none, _)))
     retried must_=== expected
   }
 
   def retryNUnitIntervalJittered = {
-    val schedule: ZSchedule[Random, Int, Int] = Schedule.recurs(5).delayed(_ => 500.millis).jittered
-    val scheduled: List[(Duration, Int)] = unsafeRun(
+    val schedule: ZSchedule[Random, Int, Int] =
+      Schedule.recurs(5).delayed(_ => 500.millis.relative).jittered
+    val scheduled: List[(Delay, Int)] = unsafeRun(
       schedule.run(List(1, 2, 3, 4, 5)).provide(TestRandom)
     )
 
-    val expected = List(1, 2, 3, 4, 5).map((250.millis, _))
-    scheduled must_=== expected
+    val expected = List(1, 2, 3, 4, 5).map((250.millis.relative, _))
+
+    val results = scheduled.zip(expected).map {
+      case (r1, r2) =>
+        unsafeRun(for {
+          d1 <- r1._1.run
+          d2 <- r2._1.run
+        } yield (d1, d2))
+    }
+
+    results.map { case (d1, d2) => d1 must_=== d2 }
   }
 
   def retryNCustomIntervalJittered = {
-    val schedule: ZSchedule[Random, Int, Int] = Schedule.recurs(5).delayed(_ => 500.millis).jittered(2, 4)
-    val scheduled: List[(Duration, Int)] = unsafeRun(
+    val schedule: ZSchedule[Random, Int, Int] =
+      Schedule.recurs(5).delayed(_ => 500.millis.relative).jittered(2, 4)
+    val scheduled: List[(Delay, Int)] = unsafeRun(
       schedule.run(List(1, 2, 3, 4, 5)).provide(TestRandom)
     )
 
-    val expected = List(1, 2, 3, 4, 5).map((1500.millis, _))
-    scheduled must_=== expected
+    val expected = List(1, 2, 3, 4, 5).map((1500.millis.relative, _))
+
+    val results = scheduled.zip(expected).map {
```

Comment:
Push this unsafe run to the top, by making a top-level for comprehension. Each test should have just 1 unsafeRun.